tests/main.test.ts

```diff
@@ -47,22 +47,66 @@ describe("src/main.ts", function() {
 
     const validAny = validModel(anyType);
 
-    it('should allow all types when no restriction', function(done) {
+    it('should allow all types for -any- schema when there are no restriction',
+    function(done) {
       const validTypes = [
         undefined,
         null,
         5,
         "string",
         true,
-        {first: 1, second: "two"},
-        // null,
-        // undefined
+        {first: 1, second: "two"}
       ];
 
-      mochaAssertPromiseResovles(
-        Promise.all(validTypes.map(validAny)),
+      mochaAssertPromiseResovles(Promise.all(validTypes.map(validAny)), done);
+    });
+
+    const anyWithBoolRestriction: anySchema = {
+      isAny: true,
+      nullAllowed: true,
+      undefinedAllowed: true,
+      restriction: (someBool: boolean) => {
+        if(someBool) {
+          return Promise.reject("error");
```

Comment:
NIT: Nicer to name "restriction-error" or anything to let us know. Same for in the test where you check for "error".